### PR TITLE
Implement temperament mappings and modals

### PR DESCRIPTION
**Pull request by frostburn**
Add temperaments dependency
Implement a prime mapping class suitable for tempering
Implement Rank 1 scale creation modal
Implement Rank 2 scale creation modal with support for temperaments built from vals and commas
Implement tempering modal for scale modification

### CODE REVIEW

**Comment by frostburn** [comment]
This depends on some new features in `moment-of-symmetry` and has overlap with #150. Merge that one first to make this commit cleaner.

File: src/components/modals/tempering-state.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
+// Component state used by RankOne, RankTwo and TemperScale
+
+import { computedAndError, fractionToString } from "@/utils";
+import { Subgroup, type TuningOptions } from "temperaments";
+import { computed, ref, watch, type Ref } from "vue";
+
+// Split text values separated by spaces, pipes, amps, semicolons or commas.
+export function splitText(text: string) {
+  return text
+    .replace(/ /g, ",")
```
Comment (frostburn) [inline src/components/modals/tempering-state.ts:10]: Any whitespace should be valid as separators.

**Comment by frostburn** [comment]
Moved the number of periods next to the period and the scale size next to MOS sizes in the Rank 2 modal.

File: src/components/modals/generation/RankTwo.vue [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+import { makeState } from "@/components/modals/tempering-state";
+import { computedAndError } from "@/utils";
+import ScaleLineInput from "@/components/ScaleLineInput.vue";
+import { Interval } from "@/interval";
+
+const props = defineProps<{
+  centsFractionDigits: number;
+}>();
+
+const MAX_SIZE = 128;
```
Comment (frostburn) [inline src/components/modals/generation/RankTwo.vue:25]: The max size is not respected when the number of periods is more than one. Need to fix in `moment-of-symmetry`.

File: src/components/modals/generation/RankOne.vue [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+          <label for="val">Val</label>
+          <input
+            ref="valElement"
+            id="val"
+            placeholder="17c"
+            class="control"
+            @focus="error = ''"
+            v-model="valsString"
+          />
+          <label for="subgroup">Subgroup</label>
```
Comment (frostburn) [inline src/components/modals/generation/RankOne.vue:79]: This could be called *Subgroup / Prime limit* to indicate the shorthand.

**Comment by frostburn** [comment]
Fixed the issues that came up during review and Discord discussion.

**Review by SeanArchibald — APPROVED** [review]

**Pull request merged**